src/rust/en.asurascans/src/lib.rs

```diff
@@ -299,31 +299,41 @@ fn get_chapter_list(manga_id: String) -> Result<Vec<Chapter>> {
 fn get_page_list(manga_id: String, chapter_id: String) -> Result<Vec<Page>> {
 	let url = get_chapter_url(&chapter_id, &manga_id);
 
-	let html = Request::new(url, HttpMethod::Get).html()?;
+	let html_text = Request::new(url.clone(), HttpMethod::Get).string()?;
 
 	let mut pages: Vec<Page> = Vec::new();
 
-	for node in html.select("div > img[alt^=chapter page]").array() {
-		let node = node.as_node()?;
-
-		let url = node.attr("abs:src").read();
-		let index = {
-			let before = url.substring_after_last('/').unwrap_or("");
-			let after = before.substring_before('.').unwrap_or("");
-
-			let cleaned_after = after
-				.chars()
-				.filter(|c| c.is_ascii_digit())
-				.collect::<String>();
-
-			cleaned_after.parse::<i32>().unwrap_or(-1)
-		};
-
-		pages.push(Page {
-			index,
-			url,
-			..Default::default()
-		});
+  let mut text_slice = html_text.as_str();
+  loop {
+    let chap = text_slice.find("https://gg.asuracomic.net/storage/media/");
+    if let Some(chap) = chap {
+      text_slice = &text_slice[chap..];
+      let end = text_slice.find(".webp").unwrap();
```

Comment:
don't use `unwrap`, if it doesn't exist either continue or return early with an error